qp_qiime2/qiime2.py

```diff
@@ -59,3 +64,93 @@ def rarefy(qclient, job_id, parameters, out_dir):
+    if not exists(out_dir):
+        mkdir(out_dir)
+
+    qclient.update_job_step(job_id, "Step 1 of 4: Collecting information")
+    artifact_id = parameters['i-table']
+    metric = parameters['p-metric']
+    tree = parameters['i-tree']
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    biom_fpi = artifact_info['files']['biom'][0]
+    biom_qza = join(out_dir, 'q2-biom.qza')
+
+    qclient.update_job_step(
+        job_id, "Step 2 of 4: Converting Qiita artifacts to Q2 artifact")
+    # converting biom
+    cmd = ('qiime tools import --input-path %s --output-path %s '
+           '--type "FeatureTable[Frequency]' % (biom_fpi, biom_qza))
+    b = load_table(biom_fpi)
+    counts = list(map(sum, b.iter_data()))
+    if min(counts) == max(counts):
+        cmd += " % Properties(['uniform-sampling'])\""
+    else:
+        cmd += '"'
+    std_out, std_err, return_value = system_call(cmd)
+    if return_value != 0:
+        error_msg = ("Error converting biom:\nStd out: %s\nStd err: %s"
+                     % (std_out, std_err))
+        return False, None, error_msg
+    # converting tree
+    if tree is not None:
+        qza_tree = join(out_dir, 'tree.qza')
+        cmd = ('qiime tools import --input-path %s --type Phylogeny[Rooted] '
+               '--output-path %s' % (tree, qza_tree))
+        tree = qza_tree
+        std_out, std_err, return_value = system_call(cmd)
+        if return_value != 0:
+            error_msg = ("Error converting biom:\nStd out: %s\nStd err: %s"
+                         % (std_out, std_err))
+            return False, None, error_msg
+
+    qclient.update_job_step(
+        job_id, "Step 3 of 4: Calculating beta diversity: %s" % (metric))
+    dtx_fp = join(out_dir, '%s.qza' % metric)
+    if tree is not None and metric in STATE_UNIFRAC_METRICS:
+        cmd = ('qiime state-unifrac %s --i-table %s --i-phylogeny %s '
+               '--o-distance-matrix %s' % (metric, biom_qza, tree, dtx_fp))
+    else:
+        cmd = ('qiime diversity beta --i-table %s --p-metric %s '
+               '--o-distance-matrix %s' % (biom_qza, metric, dtx_fp))
+    std_out, std_err, return_value = system_call(cmd)
+    if return_value != 0:
+        error_msg = ("Error in beta div %s:\nStd out: %s\nStd err: %s"
+                     % (metric, std_out, std_err))
+        return False, None, error_msg
+
+    qclient.update_job_step(
+        job_id, "Step 4 of 4: Converting Q2 to Qiita artifacts")
+    fdir = join(out_dir, 'dtx')
+    ffp = join(fdir, 'distance-matrix.tsv')
+    cmd = "qiime tools export --output-dir %s %s" % (fdir, dtx_fp)
+    std_out, std_err, return_value = system_call(cmd)
+    if return_value != 0:
+        error_msg = ("Error in Q2 -> Qiita conversion:\nStd out: "
+                     "%s\nStd err: %s" % (std_out, std_err))
+        return False, None, error_msg
+
+    ainfo = [ArtifactInfo('%s distance matrix' % metric, 'distance_matrix',
```

Comment:
`'%s distance matrix' % metric` -> `distance-matrix`. This needs to match the value in [here](https://github.com/qiita-spots/qp-qiime2/pull/4/files#diff-64e4811d769ef7e8391212a91ab4b5b0R50).